app/src/main/java/de/westnordost/streetcomplete/settings/SettingsFragment.java

```diff
@@ -47,6 +62,30 @@
 			getFragmentActivity().setCurrentFragment(new QuestSelectionFragment());
 			return true;
 		});
+
+		Preference questsInvalidation = getPreferenceScreen().findPreference("quests.invalidation");
+		questsInvalidation.setOnPreferenceClickListener(preference ->
+		{
+			new AlertDialogBuilder(getContext())
+					.setMessage(R.string.invalidation_dialog_message)
+					.setPositiveButton(R.string.invalidate_shown_tile, (dialog, which) -> {
+						if (getShownTile() != null)
+						{
+							downloadedTilesDao.remove(getShownTile());
+							Toast.makeText(getContext(), R.string.invalidation_success, Toast.LENGTH_SHORT).show();
+						} else
+						{
+							Toast.makeText(getContext(), R.string.invalidation_error, Toast.LENGTH_SHORT).show();
+						}
```

Comment:
Please leave out the success and error messages. This will only confuse the users. Instead, just check if getShownTile() is null before building the dialog and only add that answer option ("Only this tile") if the current shown tile can be determined.

Reply:
But if the invalidation succeeded, there should really be **no** success message?